src/main/java/teammates/ui/webapi/CreateCourseAction.java

```diff
@@ -32,13 +33,24 @@ void checkSpecificAccessControl() throws UnauthorizedAccessException {
 
         String institute = getNonNullRequestParamValue(Const.ParamsNames.INSTRUCTOR_INSTITUTION);
 
-        List<InstructorAttributes> existingInstructors = logic.getInstructorsForGoogleId(userInfo.getId());
+        // SQL Logic
+        List<Instructor> existingInstructors = sqlLogic.getInstructorsForGoogleId(userInfo.getId());
         boolean canCreateCourse = existingInstructors
+                .stream()
+                .filter(Instructor::hasCoownerPrivileges)
+                .map(instructor -> instructor.getCourse())
+                .filter(Objects::nonNull)
+                .anyMatch(course -> institute.equals(course.getInstitute()));
```

Comment:
should this logic be in InstructorsLogic?

Reply:
Makes sense, will move fetching of instructors to logic layer as well.